visualization/app/codeCharta/state/metric.service.ts

```diff
@@ -128,13 +132,11 @@ export class MetricService implements FilesSelectionSubscriber, BlacklistSubscri
 		return metricData.sort((a, b) => (a.name > b.name ? 1 : b.name > a.name ? -1 : 0))
 	}
 
-	private addUnaryMetric() {
-		if (!this.metricData.some(x => x.name === MetricService.UNARY_METRIC)) {
```

Comment:
I tried to figure out why this is unnecessary but I failed while just looking at the code from this file. Would you mind elaborating?

Reply:
After reading the available metrics from the cc.json we manually add the unary-metric before finally setting it in our state. Since we calculate the available metrics all together again, there is no way that this can include the unary metric.

I don't know why this piece of code was there in the first place, it might be just something we needed in the past and developers were too afraid to remove.